android/java/org/chromium/chrome/browser/vpn/activities/BraveVpnParentActivity.java

```diff
@@ -265,8 +267,16 @@ public void run() {
                             braveVpnWireguardProfileCredentials.getClientId());
                     mBraveVpnPrefModel.setApiAuthToken(
                             braveVpnWireguardProfileCredentials.getApiAuthToken());
-                    BraveVpnPrefUtils.setPrefModel(mBraveVpnPrefModel);
-
+                    CompletableFuture<Void> awaitPrefDone = new CompletableFuture<Void>();
+                    BraveVpnParentActivity.this.runOnUiThread(
+                            () -> {
+                                try {
+                                    BraveVpnPrefUtils.setPrefModel(mBraveVpnPrefModel);
+                                } finally {
+                                    awaitPrefDone.complete(null);
+                                }
+                            });
+                    awaitPrefDone.get();
```

Comment:
that function is a bit mess as it executes on background thread and in the middle in needs a UI thread and after that again bg thread. In the best case we would refactor it somehow, but it's out of scope of that task.  I would probably create another function where put everything below and call `PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK` to run the rest on the bg thread, not sure does it need to run in the bg thread. That way we remove `CompletableFuture<Void> awaitPrefDone = new CompletableFuture<Void>();` and avoid extra blockages of threads.